src/Common/src/System/Globalization/IdnMapping.Windows.cs

```diff
@@ -20,14 +20,22 @@ private unsafe string GetAsciiCore(char* unicode, int count)
             }
 
             // Do the conversion
-            char[] output = new char[length];
-            length = Interop.mincore.IdnToAscii(flags, unicode, count, output, length);
-            if (length == 0)
+            const int StackAllocThreshold = 512; // arbitrary limit to switch from stack to heap allocation
+            if (length < StackAllocThreshold)
             {
-                ThrowForZeroLength("unicode", SR.Argument_IdnIllegalName, SR.Argument_InvalidCharSequenceNoIndex);
+                char* output = stackalloc char[length];
+                Interop.mincore.IdnToAscii(flags, unicode, count, output, length);
```

Comment:
In all of these calls to IdnToAscii and IdnToUnicode, you're not storing the new length that's returned from the call. As a result, the returned strings will likely include garbage at the end.  You're also not checking for 0 and throwing; why?